django/contrib/postgres/operations.py

```diff
@@ -75,3 +77,66 @@ class UnaccentExtension(CreateExtension):
 
     def __init__(self):
         self.name = 'unaccent'
+
+
+class EnsureNotInTransactionMixin:
+    def _ensure_not_in_transaction(self, schema_editor):
+        if schema_editor.connection.in_atomic_block:
+            raise NotSupportedError(
+                'The %s operation cannot be executed inside a transaction '
+                '(set atomic = False on the migration).' % self.__class__.__name__
+            )
+
+
+class AddIndexConcurrently(EnsureNotInTransactionMixin, AddIndex):
+    """Create an index using Postgres's CREATE INDEX CONCURRENTLY syntax."""
+
+    atomic = False
+
+    def describe(self):
+        return 'Concurrently create index %s on field(s) %s of model %s' % (
+            self.index.name,
+            ', '.join(self.index.fields),
+            self.model_name,
+        )
+
+    def database_forwards(self, app_label, schema_editor, from_state, to_state):
+        self._ensure_not_in_transaction(schema_editor)
+
+        model = to_state.apps.get_model(app_label, self.model_name)
+        sql = str(self.index.create_sql(model, schema_editor, concurrently=True))
+        schema_editor.execute(sql)
+
+    def database_backwards(self, app_label, schema_editor, from_state, to_state):
+        self._ensure_not_in_transaction(schema_editor)
+
+        model = from_state.apps.get_model(app_label, self.model_name)
+        sql = str(self.index.remove_sql(model, schema_editor, concurrently=True))
```

Comment:
`remove_index()` accepts `concurrently`, moreover `self.allow_migrate_model()` check `self.allow_migrate_model()` check is missing, IMO we should use
```python
if self.allow_migrate_model(schema_editor.connection.alias, model):
    schema_editor.remove_index(model, self.index, concurrently=True)
```